commit 1edbfe1e4a0f5a56c90a6402e2671cab471e9a35 (refs/changes/28/2828/1)
Author: Angela Fong <angela.fong@joyent.com>
Date:   2017-10-19T22:52:06-07:00 (2 years ago)
    
    ADMINUI-2388 When packages with similar names are created, their package versions are listed in the other's versions

diff --git a/lib/packages.js b/lib/packages.js
index 66da16ca..506867e0 100644
--- a/lib/packages.js
+++ b/lib/packages.js
@@ -127,14 +127,18 @@ module.exports.list = function (req, res, next) {
                 return;
             }
             if (typeof (value) === 'string' && value.length) {
-                if (key === 'billing_id' || value === 'true' || value === 'false') {
+                if (key === 'billing_id' || key === 'name' || value === 'true' || value === 'false') {
                     if (key === 'billing_id') {
                         filter.push(sprintf('(%s=%s)', 'uuid', value));
                     } else {
                         filter.push(sprintf('(%s=%s)', key, value));
                     }
                 } else {
-                    filter.push(sprintf('(%s:caseIgnoreSubstringsMatch:=*%s*)', key, value));
+                    if (key === 'name_substring') {
+                        filter.push(sprintf('(%s:caseIgnoreSubstringsMatch:=*%s*)', 'name', value));
+                    } else {
+                        filter.push(sprintf('(%s:caseIgnoreSubstringsMatch:=*%s*)', key, value));
+                    }
                 }
                 return;
             }
diff --git a/www/js/views/packages.js b/www/js/views/packages.js
index caf76183..8799573a 100644
--- a/www/js/views/packages.js
+++ b/www/js/views/packages.js
@@ -222,7 +222,7 @@ var PackagesView = Backbone.Marionette.Layout.extend({
 
         var data = {
             active: getValue('active'),
-            name: getValue('name'),
+            name_substring: getValue('name'),
             traits: getValue('traits'),
             group: getValue('group'),
             billing_id: getValue('billing_id')
